api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -21,4 +32,56 @@ public void setDao(PatientQueueingDao dao) {
 		this.dao = dao;
 	}
 	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#savePatientQue(org.openmrs.module.patientqueueing.model.PatientQueue)
+	 */
+	public PatientQueue savePatientQue(PatientQueue patientQueue) throws Exception {
+		return dao.savePatientQueue(patientQueue);
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getPatientQueueById(java.lang.Integer)
+	 */
+	public PatientQueue getPatientQueueById(Integer queueId) throws APIException {
+		return dao.getPatientQueueById(queueId);
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getPatientQueueList(org.openmrs.Provider,
+	 *      java.util.Date, java.util.Date, org.openmrs.Location, org.openmrs.Location,
+	 *      org.openmrs.Patient, java.lang.String)
+	 */
+	public List<PatientQueue> getPatientQueueList(Provider provider, Date fromDate, Date toDate, Location locationTo,
+	        Location locationFrom, Patient patient, String status) throws APIException {
+		return dao.getPatientQueueList(provider, fromDate, toDate, locationTo, locationFrom, patient, status);
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#completePatientQueue(org.openmrs.module.patientqueueing.model.PatientQueue)
+	 */
+	@Override
+	public PatientQueue completePatientQueue(PatientQueue patientQueue) throws Exception {
+		patientQueue.setStatus(PatientQueueingConfig.status.COMPLETED.toString());
```

Comment:
Any reason why you are calling toString(), i thought the parameter in setStatus is an enum.